### PR TITLE
add secret to glossary

### PR DESCRIPTION
**Pull request by zachpuck**
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> For 1.9 Features: set Milestone to `1.9` and Base Branch to `release-1.9`
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> NOTE: Please check the “Allow edits from maintainers” box (see image below) to
> [allow reviewers to fix problems](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) on your patch and speed up the review process.
>
> Please delete this note before submitting the pull request.
>
> NOTE: After opening the PR, please *un-check and re-check* the "Allow edits from maintainers" box. This is a temporary workaround to address a known issue with GitHub.

Allow edits from maintainers checkbox

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/website/6577)

### CODE REVIEW

File: _data/glossary/secret.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+name: Secret
+full-link: /docs/concepts/configuration/secret/
+related:
+- pods
+- volume
+tags:
+- core-object
+- security
+short-description: >
+  stores sensitive information, such as passwords, OAuth tokens, and ssh keys.
```
Comment (Bradamant3) [inline _data/glossary/secret.yml]: s/stores/Stores

File: _data/glossary/secret.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+related:
+- pods
+- volume
+tags:
+- core-object
+- security
+short-description: >
+  Stores sensitive information, such as passwords, OAuth tokens, and ssh keys.
+long-description: >
+  Allows for more control over how sensistive information is used and reduces the risk of accidental exposure.
```
Comment (Bradamant3) [inline _data/glossary/secret.yml]: s/sensistive/sensitive
Mention encryption. Maybe also compare with ConfigMap? This is a nice resource: https://medium.com/google-cloud/kubernetes-configmaps-and-secrets-68d061f7ab5b
Thanks for your help!

File: _data/glossary/secret.yml (hunk 1 above)
Comment (zachpuck) [reply _data/glossary/secret.yml]: done

File: _data/glossary/secret.yml (hunk 2 above)
Comment (zachpuck) [reply _data/glossary/secret.yml]: done

**Comment by tonglil** [comment]
Secret also defined in https://github.com/kubernetes/website/pull/6581.

**Pull request merged**